backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inout/impl/HUInOutBL.java

```diff
@@ -240,9 +254,16 @@ public void destroyHUs(@NonNull final org.compiere.model.I_M_InOut inout)
 			huContext.getHUPackingMaterialsCollector().setCollectIfOwnPackingMaterialsOnly(true);
 		}
 
+		huContext.setProperty(IHUContext.PROPERTY_IsReceiptReversal, true);
 		//
 		// Mark assigned HUs as destroyed
 		handlingUnitsBL.markDestroyed(huContext, hus);
+
+		// If the HUs were linked to M_Package_HU entries, delete them too
+		for (final I_M_HU hu : hus)
+		{
+			huPackageBL.destroyHUPackages(HuId.ofRepoId(hu.getM_HU_ID()));
```

Comment:
avoid SQL N+1
pls introduce and use HUPackageBL.destroyHUPackages(Set<HuId> huIds)